### PR TITLE
Stripe checkout success

### PR DESCRIPTION
**Pull request by julia-kim**

### CODE REVIEW

File: src/main/java/com/southernartsoap/controller/CartController.java [hunk 1]
```diff
@@ -34,86 +37,98 @@
...
-//        System.out.println(cartDetails); //somehow this isn't being passed in?
-//        if(!bindingResult.hasErrors()){
-//            cartDetailsService.removeCartDetailFromCartByCartDetailsId(cartDetails.getId());
-//            
-//        }
-//    
-//        return "cart";
-//    }
-    
+		return "redirect:/cart";
```
Comment (jlangson) [inline src/main/java/com/southernartsoap/controller/CartController.java:99]: advantage of redirect over cart() ?

File: src/main/java/com/southernartsoap/model/ChargeRequest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.southernartsoap.model;
+
+import lombok.Data;
+
+@Data
```
Comment (jlangson) [inline src/main/java/com/southernartsoap/model/ChargeRequest.java:5]: shouldn't need getters and setters with lombok.

File: src/main/resources/templates/result.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <div class="container">
+      <h3 th:if="${error}" th:text="${error}"></h3>
+      <div th:unless="${error}">
+        <h3>Success!</h3>
+        <div>Id.: <span th:text="${id}" /></div>
+        <div>Status: <span th:text="${status}" /></div>
+        <div>Charge id.: <span th:text="${chargeId}" /></div>
+        <div>Balance transaction id.: <span th:text="${balance_transaction}" /></div>
+      </div>
+      <a th:href="@{/}">Check out more of our delicious soaps!</a>
```
Comment (jlangson) [inline src/main/resources/templates/result.html:18]: Is soap edible? ;p

**Review by jlangson — CHANGES_REQUESTED** [review]
remove getters and setters in ChargeRequest.java

have checkout clear the cart.

File: src/main/resources/templates/result.html (hunk 3 above)
Comment (julia-kim) [reply src/main/resources/templates/result.html:18]: haha no, but I'm bad at writing copy, and I think the owner referred to them as delicious in some of her product descriptions. got anything better?

File: src/main/java/com/southernartsoap/controller/ChargeController.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		chargeRequest.setCurrency(ChargeRequest.Currency.USD);
+		Charge charge = paymentsService.charge(chargeRequest);
+		model.addAttribute("id", charge.getId());
+		model.addAttribute("status", charge.getStatus());
+		model.addAttribute("chargeId", charge.getId());
+		model.addAttribute("balance_transaction", charge.getBalanceTransaction());
+		
+		// delete all items from user's cart
+		User user = userService.getLoggedInUser();
+		userService.deleteCart(user);
```
Comment (jlangson) [inline src/main/java/com/southernartsoap/controller/ChargeController.java:36]: this doesn't seem to delete

File: src/main/java/com/southernartsoap/service/UserService.java [hunk 5]
```diff
@@ -9,80 +8,58 @@
...
+	public void updateCart(Cart cart) {
+		User user = getLoggedInUser();
+		user.setCart(cart);
+		saveExisting(user);
+	}
 
-    public User getLoggedInUser() {
-        return findByUsername(SecurityContextHolder.getContext().getAuthentication().getName());
-    }
+	// attempt to clear cart
```
Comment (jlangson) [inline src/main/java/com/southernartsoap/service/UserService.java:51]: this doesn't seem to delete the cart

**Pull request closed without merging**